### PR TITLE
Add console endpoint config section and init arg 

### PR DESCRIPTION
**Pull request by andrewwdye**
Signed-off-by: Andrew Dye <andrewwdye@gmail.com>

# TL;DR
Console endpoint may be different from admin (as is the case for sandbox). Add a console config section and init arg to the `config init` command.

## Type
- [ ] Bug Fix
- [x] Feature
- [ ] Plugin

## Are all requirements met?

- [x] Code completed
- [x] Smoke tested
- [x] Unit tests added
- [x] Code documentation added
- [ ] Any pending items have an associated Issue

## Complete description
This change
1. Adds a new `console` config section to capture the new optional endpoint. This allows the new console endpoint to be read from the flytectl config file.
2. Adds a `--console` arg to the `config init` command. If provided the console section is written, otherwise is left blank. 
3. Populate the sandbox config file with console endpoint, to be consumed by pyflyte run in a separate PR

**Note:** that once the flytectl config file is written with the new console section, it can no longer be read by old versions of flytectl. It will fail with a message like `strict mode is on but received keys [map[console:{}]] to decode with no config assigned to receive them: failed strict mode check`

Example config file
```
admin:
  # For GRPC endpoints you might want to use dns:///flyte.myexample.com
  endpoint: dns:///localhost:30081
  authType: Pkce
  insecure: true
console:
  endpoint: http://localhost:30080
logger:
  show-source: true
  level: 0
```

## Tracking Issue
NA

## Follow-up issue
NA

### CODE REVIEW

File: cmd/config/subcommand/config/console_flags.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package config
+
+import "github.com/flyteorg/flytestdlib/config"
+
+//go:generate pflags ConsoleConfig --default-var DefaultConsoleConfig --bind-default-var
+
+var (
+	DefaultConsoleConfig = &ConsoleConfig{}
```
[the remark this answers is not in the document]
Comment (andrewwdye) [reply cmd/config/subcommand/config/console_flags.go:8]: I've modified pyflyte run to consume console.endpoint from the flytectl coming file in https://github.com/flyteorg/flytekit/pull/1233 so that it doesn't need to special case localhost:30081. 

This section is required to validate and load the config file in flytectl even though the new section won't be consumed here.

**Review by eapolinario — APPROVED** [review]
